From bcbaa293d36d28bc0eeaafe7ddf8a92d8f27b593 Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <vladimir.oltean@nxp.com>
Date: Mon, 16 Dec 2019 15:09:49 +0200
Subject: [PATCH 25/26] net: mscc: ocelot: export ANA, DEV and QSYS registers
 to include/soc/mscc

Since the Felix DSA driver is implementing its own PHYLINK instance due
to SoC differences, it needs access to the few registers that are
common, mainly for flow control.

Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>

Conflicts:
	drivers/net/ethernet/mscc/ocelot_tsn.c

which has been added in downstream patch b5c05e3404a5 ("net: mscc:
ocelot: tsn configuration support") and also needs to be adapted to the
new location of the header files.
---
 drivers/net/ethernet/mscc/ocelot.h                       | 6 +++---
 drivers/net/ethernet/mscc/ocelot_tsn.c                   | 4 ++--
 {drivers/net/ethernet => include/soc}/mscc/ocelot_ana.h  | 0
 {drivers/net/ethernet => include/soc}/mscc/ocelot_dev.h  | 0
 {drivers/net/ethernet => include/soc}/mscc/ocelot_qsys.h | 0
 5 files changed, 5 insertions(+), 5 deletions(-)
 rename {drivers/net/ethernet => include/soc}/mscc/ocelot_ana.h (100%)
 rename {drivers/net/ethernet => include/soc}/mscc/ocelot_dev.h (100%)
 rename {drivers/net/ethernet => include/soc}/mscc/ocelot_qsys.h (100%)

diff --git a/drivers/net/ethernet/mscc/ocelot.h b/drivers/net/ethernet/mscc/ocelot.h
index 4ff5b045c181..90b7601408c1 100644
--- a/drivers/net/ethernet/mscc/ocelot.h
+++ b/drivers/net/ethernet/mscc/ocelot.h
@@ -18,11 +18,11 @@
 #include <linux/ptp_clock_kernel.h>
 #include <linux/regmap.h>
 
+#include <soc/mscc/ocelot_qsys.h>
 #include <soc/mscc/ocelot_sys.h>
+#include <soc/mscc/ocelot_dev.h>
+#include <soc/mscc/ocelot_ana.h>
 #include <soc/mscc/ocelot.h>
-#include "ocelot_ana.h"
-#include "ocelot_dev.h"
-#include "ocelot_qsys.h"
 #include "ocelot_rew.h"
 #include "ocelot_qs.h"
 #include "ocelot_tc.h"
diff --git a/drivers/net/ethernet/mscc/ocelot_tsn.c b/drivers/net/ethernet/mscc/ocelot_tsn.c
index 8b52c9510d1d..7c0683b4ccad 100644
--- a/drivers/net/ethernet/mscc/ocelot_tsn.c
+++ b/drivers/net/ethernet/mscc/ocelot_tsn.c
@@ -11,8 +11,8 @@
 #include <linux/iopoll.h>
 #include "ocelot.h"
 #include <soc/mscc/ocelot_sys.h>
-#include "ocelot_ana.h"
-#include "ocelot_qsys.h"
+#include <soc/mscc/ocelot_ana.h>
+#include <soc/mscc/ocelot_qsys.h>
 #include "ocelot_rew.h"
 #include "ocelot_dev_gmii.h"
 #include "ocelot_tsn.h"
diff --git a/drivers/net/ethernet/mscc/ocelot_ana.h b/include/soc/mscc/ocelot_ana.h
similarity index 100%
rename from drivers/net/ethernet/mscc/ocelot_ana.h
rename to include/soc/mscc/ocelot_ana.h
diff --git a/drivers/net/ethernet/mscc/ocelot_dev.h b/include/soc/mscc/ocelot_dev.h
similarity index 100%
rename from drivers/net/ethernet/mscc/ocelot_dev.h
rename to include/soc/mscc/ocelot_dev.h
diff --git a/drivers/net/ethernet/mscc/ocelot_qsys.h b/include/soc/mscc/ocelot_qsys.h
similarity index 100%
rename from drivers/net/ethernet/mscc/ocelot_qsys.h
rename to include/soc/mscc/ocelot_qsys.h
-- 
2.17.1

